HakawaiTests/HKWMentionsPluginTests.m

```diff
@@ -362,20 +378,70 @@ - (BOOL)stringValidForMentionsCreation:(NSString *)string;
         m1.range = NSMakeRange(0, m1.mentionText.length);
 
         [mentionsPlugin addMention:m1];
-        dispatch_async(dispatch_get_main_queue(), ^{
-            expect(mentionsPlugin.mentions.count).to.equal(1);
-        });
+        expect(mentionsPlugin.mentions.count).to.equal(1);
 
         // delete the whole mention
         BOOL deletionResult2 = [mentionsPlugin textView:textView shouldChangeTextInRange:NSMakeRange(m1.mentionText.length-1, 1) replacementText:@""];
-        dispatch_async(dispatch_get_main_queue(), ^{
-            expect(deletionResult2).to.equal(NO);
-            expect(mentionsPlugin.mentions.count).to.equal(0);
-            expect([textView.text length]).to.equal(0);
-        });
+        expect(deletionResult2).to.equal(NO);
+        expect(mentionsPlugin.mentions.count).to.equal(0);
+        expect([textView.text length]).to.equal(0);
+    });
+
+    it(@"should properly handle range mention deletion", ^{
+        HKWMentionsAttribute *m1 = [HKWMentionsAttribute mentionWithText:@"Asdf ghjkl" identifier:@"1"];
+        HKWMentionsAttribute *m2 = [HKWMentionsAttribute mentionWithText:@"Asdf ghjkll" identifier:@"2"];
 
+        expect(mentionsPlugin.mentions.count).to.equal(0);
 
+        [textView insertText:m1.mentionText];
+        NSString *string = @" Hello ";
+        [textView insertText:string];
+        [textView insertText:m2.mentionText];
+        m1.range = NSMakeRange(0, m1.mentionText.length);
+        m2.range = NSMakeRange(m1.mentionText.length + string.length, m2.mentionText.length);
+
+        [mentionsPlugin addMention:m1];
+        [mentionsPlugin addMention:m2];
+        expect(mentionsPlugin.mentions.count).to.equal(2);
+
+        // delete the whole mention
+        NSUInteger middleOfMention1 = m1.mentionText.length/2;
+        BOOL deletionResult2 = [mentionsPlugin textView:textView shouldChangeTextInRange:NSMakeRange(middleOfMention1, m1.mentionText.length+string.length+m2.mentionText.length/2-middleOfMention1) replacementText:@""];
+        expect(deletionResult2).to.equal(NO);
+        expect(mentionsPlugin.mentions.count).to.equal(0);
+        expect([textView.text length]).to.equal(0);
     });
+
+    it(@"should properly handle range mention deletion with trimming", ^{
```

Comment:
can we make sure all those deletion cases are tested, would be helpful to clarify those cases in the test description too:
1) cursor range before a mention
2) cursor range after a mention
3) cursor range within a mention range and mention range within a cursor range
4) cursor range intercepts first half of a mention
5) cursor range intercepts second half of a mention